gflownet/gflownet.py

```diff
@@ -242,14 +247,22 @@ def sample_actions(
             envs = [envs]
         # Build states and masks
         states = [env.state for env in envs]
-        if is_forward:
-            mask_invalid_actions = tbool(
-                [env.get_mask_invalid_actions_forward() for env in envs],
-                device=self.device,
-            )
+        if mask_invalid_actions is None:
+            # Invalid actions masks are not provided, they need to be computed
+            if is_forward:
+                mask_invalid_actions = tbool(
+                    [env.get_mask_invalid_actions_forward() for env in envs],
+                    device=self.device,
+                )
+            else:
+                mask_invalid_actions = tbool(
+                    [env.get_mask_invalid_actions_backward() for env in envs],
+                    device=self.device,
+                )
         else:
+            # Invalid action masks are provided, convert to pytorch tensors
             mask_invalid_actions = tbool(
-                [env.get_mask_invalid_actions_backward() for env in envs],
+                [m for m in mask_invalid_actions],
```

Comment:
Why do you need to iterate over `mask_invalid_actions` instead of converting calling `tbool` directly on it?
```suggestion
                mask_invalid_actions,
```
(the suggestion would need a `black` though.

Reply:
Ha! I was definitely half asleep here. You're absolutely right, I'll change that.

Reply:
Done.